pkg/crdpuller/discovery.go

```diff
@@ -105,8 +106,23 @@ func (sp *schemaPuller) PullCRDs(context context.Context, resourceNames ...strin
 		return nil, err
 	}
 
+	groupResources, err := restmapper.GetAPIGroupResources(sp.discoveryClient)
+	if err != nil {
+		return nil, err
+	}
+	mapper := restmapper.NewDiscoveryRESTMapper(groupResources)
+
 	pullAllResources := len(resourceNames) == 0
-	resourcesToPull := sets.NewString(resourceNames...)
+	resourcesToPull := sets.NewString()
+	for _, resourceToPull := range resourceNames {
+		gr := schema.ParseGroupResource(resourceToPull)
+		gvr, err := mapper.ResourceFor(gr.WithVersion(""))
```

Comment:
If we have versions in discovery, why will this be valid?

Reply:
RESTMappers perform matches against partial data. This is e.g. how kubectl works when you do `kubectl get po` or `kubectl get deployments` - it uses a RESTMapper to translate a partial group+resource specification (po=pods (.core), deployments = deployments.apps) to the highest priority/preferred GVR, as specified by the server, assuming it can find a match.

Reply:
I think for the time being this is fine. It originates from the over-simplified flag `--resources-to-sync` which takes `pods`, `configmaps`, etc. In the future we will likely drive this through some API and then we will fully qualify the list of resources.

Reply:
And I would add that in line [137](https://github.com/kcp-dev/kcp/pull/366/files#diff-2846812f7811365a5002800bbdde45978a15c09070f64b90eafcd5f72bb5e4d0R137) we only get the preferred version for each `GroupResource` by calling `ServerPreferredResources`.